ipa-multipoint/ipa_multipoint_jni/src/lib.rs

```diff
@@ -45,130 +39,92 @@ pub extern "system" fn Java_org_hyperledger_besu_nativelib_ipamultipoint_LibIpaM
-    scalar_bytes.reverse();
-
-    return env.byte_array_from_slice(&scalar_bytes).expect("Couldn't convert to byte array");
-}
+pub extern "system" fn Java_org_hyperledger_besu_nativelib_ipamultipoint_LibIpaMultipoint_commit(
+    env: JNIEnv,
+    _class: JClass<'_>,
+    input: jbyteArray,
+) -> jbyteArray {
+    let mut input = env
+        .convert_byte_array(input)
+        .expect("Cannot convert jbyteArray to rust array");
 
+    let committer = &CONFIG.committer;
+
+    reverse_scalars_endian(&mut input);
+    let commitment = ffi_interface::commit_to_scalars(committer, &input).unwrap();
+
+    let mut hash = ffi_interface::hash_commitment(commitment);
+    // TODO(Big-Endian): The output of `hash_commitment` is a scalar serialized
+    // TODO(Big-Endian): in little endian. The previous implementation used big endian.
+    hash.reverse();
+
+    env.byte_array_from_slice(&hash)
+        .expect("Couldn't convert to byte array")
+}
 
 /// Commit_root receives a list of 32 byte scalars and returns a 32 byte commitment.to_bytes()
 /// This is ported from rust-verkle.
 #[no_mangle]
-pub extern "system" fn Java_org_hyperledger_besu_nativelib_ipamultipoint_LibIpaMultipoint_commitRoot(env: JNIEnv,
-                                                                                                 _class: JClass<'_>,
-                                                                                                 input: jbyteArray)
-                                                                                                 -> jbyteArray {
-    // Input should be a multiple of 32-be-bytes.
-    let inp = env.convert_byte_array(input).expect("Cannot convert jbyteArray to rust array");
-    let len = inp.len();
-    if len % 32 != 0 {
-        env.throw_new("java/lang/IllegalArgumentException", "Invalid input length. Should be a multiple of 32-bytes.")
-           .expect("Failed to throw exception");
-        return std::ptr::null_mut(); // Return null pointer to indicate an error
-    }    
-    let n_scalars = len / 32;
-    if n_scalars > 256 {
-        env.throw_new("java/lang/IllegalArgumentException", "Invalid input length. Should be at most 256 elements of 32-bytes.")
-           .expect("Failed to throw exception");
-        return std::ptr::null_mut(); // Return null pointer to indicate an error
-    }    
-
-    // Each 32-be-bytes are interpreted as field elements.
-    let mut scalars: Vec<Fr> = Vec::with_capacity(n_scalars);
-    for b in inp.chunks(32) {
-        scalars.push(Fr::from_be_bytes_mod_order(b));
-    }
-    
-    // Committing all values at once.
-    let bases = CRS::new(n_scalars, PEDERSEN_SEED);
-    let commit = multi_scalar_mul(&bases.G, &scalars);
+pub extern "system" fn Java_org_hyperledger_besu_nativelib_ipamultipoint_LibIpaMultipoint_commitRoot(
+    env: JNIEnv,
+    _class: JClass<'_>,
+    input: jbyteArray,
+) -> jbyteArray {
+    let mut input = env
```

Comment:
Same comment, we should have a uniform way of converting inputs and outputs from/to Java.